W3ChampionsStatisticService/Matches/OngoingMatchesCache.cs

```diff
@@ -20,12 +92,28 @@ public async Task<long> CountOnGoingMatches(
         int minMmr,
         int maxMmr)
     {
-        await UpdateCacheIfNeeded();
-        return _values.Count(m => (gameMode == GameMode.Undefined || m.GameMode == gameMode)
-                                    && (gateWay == GateWay.Undefined || m.GateWay == gateWay)
-                                    && (map == "Overall" || m.Map == map)
-                                    && (minMmr == 0 || !m.Teams.Any(team => team.Players.Any(player => player.OldMmr < minMmr)))
-                                    && (maxMmr == 3000 || !m.Teams.Any(team => team.Players.Any(player => player.OldMmr > maxMmr))));
+        var cacheKey = string.Join("|", gameMode, gateWay, map, minMmr, maxMmr);
+        if (_countOngoingMatchesCache.Get(cacheKey) is CacheResult<long> cachedValue)
+        {
+            if (cachedValue.IsNegativeCache)
+            {
+                return 0L;
+            }
+
+            return cachedValue.Value;
+        }
+
+        await EnsureCachesInitialized();
```

Comment:
Why are we ensuring cache initialization after reading from a cache? I know it doesn't init the count cache, but it still reads weirdly

Reply:
Yeah it was more about keeping the read and the write paths separate, but I can see that one might want to have it all the start. I don't really like th seeding of the cache anyway tbh :/

Would like to have a proper init method :/